v5.0.3 配置 DNS 后 panic

<!--
除非特殊情况，请完整填写所有问题。不按模板发的 issue 将直接被关闭。
如果你遇到的问题不是 V2Ray 的 bug，比如你不清楚如何配置，请在 https://github.com/v2fly/v2ray-core/discussions 进行讨论。
-->

## 你正在使用哪个版本的 V2Ray？
v5.0.3, linux arm64
<!-- 如果服务端和客户端使用了不同版本，请注明 -->


## 你的使用场景是什么？
本地测试
<!-- 比如使用 Chrome 通过 Socks/VMess 代理观看 YouTube 视频 -->


## 你看到的异常现象是什么？
添加了 DNS 配置项时 V2Ray 无法启动，删除 DNS 配置后正常
<!-- 请描述具体现象，比如访问超时、TLS 证书错误等 -->


## 你期待看到的正常表现是怎样的？
正常使用 DNS 功能


## 请附上你的配置

<!-- 提交 issue 前，请隐去服务器域名或 IP 地址 -->

**服务端配置：**

```javascript
// 在这里附上服务器端配置文件

```

**客户端配置：**

```javascript
// 在这里附上客户端配置
{
  "inbounds": [
    {
      "tag": "socks_in",
      "port": 10086,
      "listen": "0.0.0.0",
      "protocol": "socks",
      "settings": null,
      "sniffing": {
        "enabled": true,
        "destOverride": [
          "http",
          "tls"
        ]
      }
    }
  ],
  "outbounds": [
    {
      "tag": "proxy_tcp",
      "protocol": "vmess",
      "settings": {
        "address": "127.0.0.1",
        "port": 10000,
        "uuid": "4d8ba710-e4f2-3520-8eaf-4420ed86aca7"
      }
    }
  ],
  "dns": {
    "nameServer": [
      {
        "address": {
          "network": "UDP",
          "address": "1.1.1.1",
          "port": 53
        },
        "prioritizedDomain": [
          {
            "type": "Subdomain",
            "domain": "google.com"
          }
        ]
      }
    ]
  }
}

```

## 请附上出错时软件输出的错误日志

<!-- 在 Linux 中，日志通常在 `/var/log/v2ray/error.log` 文件中 -->

**服务器端错误日志：**

```javascript
// 在这里附上服务器端日志

```

**客户端错误日志：**

```javascript
// 在这里附上客户端日志

```

## 请附上访问日志

<!-- 在 Linux 中，访问日志通常在 `/var/log/v2ray/access.log` 文件中 -->

```javascript
// 在这里附上服务器端日志

```

## 其它相关的配置文件（如 Nginx）和相关日志



## 如果 V2Ray 无法启动，请附上 `--test` 命令的输出



## 如果 V2Ray 服务运行异常，请附上 journal 日志
V2Ray 5.0.3 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.17.6 linux/arm64)
A unified platform for anti-censorship.
panic: V is not in context.

goroutine 1 [running]:
github.com/v2fly/v2ray-core/v5.MustFromContext({0xcc2aa0, 0x400010ae10})
        github.com/v2fly/v2ray-core/v5/context.go:24 +0x94
github.com/v2fly/v2ray-core/v5.RequireFeatures({0xcc2aa0, 0x400010ae10}, {0x9be0a0, 0x40004f3f40})
        github.com/v2fly/v2ray-core/v5/v2ray.go:152 +0x30
github.com/v2fly/v2ray-core/v5/app/dns.NewClient({0xcc2aa0, 0x400010ae10}, 0x400012f360, {0x0, 0x0, 0x0}, {{0x0, 0x0, 0x0}}, 0x4000197428, ...)
        github.com/v2fly/v2ray-core/v5/app/dns/nameserver.go:74 +0x188
github.com/v2fly/v2ray-core/v5/app/dns.New({0xcc2aa0, 0x400010ae10}, 0x400057c370)
        github.com/v2fly/v2ray-core/v5/app/dns/dns.go:129 +0x780
github.com/v2fly/v2ray-core/v5/app/dns.init.1.func1({0xcc2aa0, 0x400010ae10}, {0xaf71e0, 0x400057c370})
        github.com/v2fly/v2ray-core/v5/app/dns/dns.go:315 +0x4c
github.com/v2fly/v2ray-core/v5/common.CreateObject({0xcc2aa0, 0x400010ae10}, {0xaf71e0, 0x400057c370})
        github.com/v2fly/v2ray-core/v5/common/type.go:34 +0x1ec
github.com/v2fly/v2ray-core/v5/app/dns.init.1.func2({0xcc2aa0, 0x400010ade0}, {0xaeebe0, 0x400057ae10})
        github.com/v2fly/v2ray-core/v5/app/dns/dns.go:380 +0x6e0
github.com/v2fly/v2ray-core/v5/common.CreateObject({0xcc2aa0, 0x400010ade0}, {0xaeebe0, 0x400057ae10})
        github.com/v2fly/v2ray-core/v5/common/type.go:34 +0x1ec
github.com/v2fly/v2ray-core/v5.CreateObject(0x40004f3e50, {0xaeebe0, 0x400057ae10})
        github.com/v2fly/v2ray-core/v5/functions.go:19 +0x6c
github.com/v2fly/v2ray-core/v5.initInstanceWithConfig(0x4000289290, 0x40004f3e50)
        github.com/v2fly/v2ray-core/v5/v2ray.go:194 +0x14c
github.com/v2fly/v2ray-core/v5.New(0x4000289290)
        github.com/v2fly/v2ray-core/v5/v2ray.go:162 +0x88
github.com/v2fly/v2ray-core/v5/main/commands.startV2Ray()
        github.com/v2fly/v2ray-core/v5/main/commands/run.go:205 +0x324
github.com/v2fly/v2ray-core/v5/main/commands.executeRun(0x13bc6c0, {0x4000182140, 0x4, 0x4})
        github.com/v2fly/v2ray-core/v5/main/commands/run.go:82 +0xbc
github.com/v2fly/v2ray-core/v5/main/commands/base.Execute()
        github.com/v2fly/v2ray-core/v5/main/commands/base/execute.go:64 +0x6d4
main.main()
        github.com/v2fly/v2ray-core/v5/main/main.go:16 +0x300

<!-- 通常的命令为 `journalctl -u v2ray` -->

<!-- 请预览一下你填写的内容并整理好格式后，再提交 -->

--- a/app/dns/dns.go
+++ b/app/dns/dns.go
@@ -315,7 +315,7 @@ func init() {
 		return New(ctx, config.(*Config))
 	}))
 
-	common.Must(common.RegisterConfig((*SimplifiedConfig)(nil), func(ctx context.Context, config interface{}) (interface{}, error) { // nolint: staticcheck
+	common.Must(common.RegisterConfig((*SimplifiedConfig)(nil), func(ctx context.Context, config interface{}) (interface{}, error) {
 		ctx = cfgcommon.NewConfigureLoadingContext(ctx)
 
 		geoloadername := platform.NewEnvFlag("v2ray.conf.geoloader").GetValue(func() string {

--- a/app/dns/dns.go
+++ b/app/dns/dns.go
@@ -316,7 +316,7 @@ func init() {
 	}))
 
 	common.Must(common.RegisterConfig((*SimplifiedConfig)(nil), func(ctx context.Context, config interface{}) (interface{}, error) { // nolint: staticcheck
-		ctx = cfgcommon.NewConfigureLoadingContext(context.Background()) // nolint: staticcheck
+		ctx = cfgcommon.NewConfigureLoadingContext(ctx)
 
 		geoloadername := platform.NewEnvFlag("v2ray.conf.geoloader").GetValue(func() string {
 			return "standard"

